File: 백준/Silver/1543. 문서 검색/문서 검색.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const document = input[0].split("")
const visited = Array.from({ length: document.length }, () => false)
const word = input[1]
let cnt = 0


for (let i = 0; i < document.length; i++) {
  if (!visited[i] && document.slice(i, i + word.length).join("") === word) {
    for (let j = 0; j < word.length; j++) {
      visited[i + j] = true
    }
    cnt += 1
  }
}

console.log(cnt)

File: 백준/Silver/1543. 문서 검색/문서 검색.py
import sys

input = sys.stdin.read().strip().split("\n")

document = list(input[0])
visited = [False] * len(document)
word = input[1]
cnt = 0

for i in range(len(document)):
    if not visited[i] and ''.join(document[i:i + len(word)]) == word:
        for j in range(len(word)):
            visited[i + j] = True
        cnt += 1

print(cnt)
